### PR TITLE
Fix unit tests with PHPUnit 9 errors

### LINKED ISSUE #29329: [Issue] Fix unit tests with PHPUnit 9 errors
This issue is automatically created based on existing pull request: magento/magento2#29244: Fix unit tests with PHPUnit 9 errors

---------

### Description (*)

Most of the failing tests are due to:
- Depending on setUp logic in dataProviders, while dataProvider code is called before setUp
- Trying to mock non-existing methods with createPartialMock which is prohibited in PHPUnit 9

### Fixed Issues (if relevant)

1. Fixes magento/magento2#<issue_number>

### Manual testing scenarios (*)

1. ...
2. ...

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds are green)

### PR DESCRIPTION
**Pull request by AntonEvers**
### Description (*)

Most of the failing tests are due to:
- Depending on setUp logic in dataProviders, while dataProvider code is called before setUp
- Trying to mock non-existing methods with createPartialMock which is prohibited in PHPUnit 9

### Fixed Issues (if relevant)

1. Fixes magento/magento2#<issue_number>

### Manual testing scenarios (*)

1. ...
2. ...

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds are green)

### Resolved issues:
1. [x] resolves magento/magento2#29329: Fix unit tests with PHPUnit 9 errors

### CODE REVIEW

**Comment by AntonEvers** [comment]
@magento run all tests

**Comment by AntonEvers** [comment]
@ihor-sviziev can I ask for your review on this? Starting from today I saw a lot of failing tests.

**Review by ihor-sviziev — DISMISSED** [review]

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-7874](https://jira.corp.magento.com/browse/ENGCOM-7874) has been created to process this Pull Request

**Comment by engcom-Alfa** [comment]
QA not applicable

File: app/code/Magento/Quote/Test/Unit/Model/Quote/Address/Total/SubtotalTest.php [hunk 1]
```diff
@@ -110,10 +110,14 @@ public function testCollect($price, $originalPrice, $itemHasParent, $expectedPri
...
         );
         /** @var Address|MockObject $address */
-        $address = $this->createPartialMock(
-            Address::class,
-            ['setTotalQty', 'getTotalQty', 'removeItem', 'getQuote']
-        );
+        $address = $this->getMockBuilder(Address::class)
+            ->disableOriginalConstructor()
+            ->disableOriginalClone()
+            ->disableArgumentCloning()
```
Comment (lbajsarowicz) [inline app/code/Magento/Quote/Test/Unit/Model/Quote/Address/Total/SubtotalTest.php]: **YAGNI** If original code is not performing any `clone` then you don't need to provide these methods.

File: app/code/Magento/Quote/Test/Unit/Model/Quote/Address/Total/SubtotalTest.php [hunk 2]
```diff
@@ -161,10 +165,13 @@ public function testCollect($price, $originalPrice, $itemHasParent, $expectedPri
...
 
-        $total = $this->createPartialMock(
-            Total::class,
-            ['setBaseVirtualAmount', 'setVirtualAmount']
-        );
+        $total = $this->getMockBuilder(Total::class)
+            ->disableOriginalConstructor()
+            ->disableOriginalClone()
+            ->disableArgumentCloning()
+            ->disallowMockingUnknownTypes()
```
Comment (lbajsarowicz) [inline app/code/Magento/Quote/Test/Unit/Model/Quote/Address/Total/SubtotalTest.php]: **YAGNI** If original code is not performing any `clone` then you don't need to provide these methods.

File: app/code/Magento/Quote/Test/Unit/Model/Quote/Address/Total/SubtotalTest.php [hunk 3]
```diff
@@ -229,13 +239,14 @@ public function testCollectWithInvalidItems()
...
-            [
-                'getId',
-                'getQuoteItemId'
-            ]
-        );
+        $addressItem = $this->getMockBuilder(AddressItem::class)
+            ->disableOriginalConstructor()
+            ->disableOriginalClone()
+            ->disableArgumentCloning()
+            ->disallowMockingUnknownTypes()
```
Comment (lbajsarowicz) [inline app/code/Magento/Quote/Test/Unit/Model/Quote/Address/Total/SubtotalTest.php]: **YAGNI** If original code is not performing any `clone` then you don't need to provide these methods.

File: app/code/Magento/Sales/Test/Unit/Model/Order/CreditmemoFactoryTest.php [hunk 4]
```diff
@@ -49,10 +49,14 @@ class CreditmemoFactoryTest extends TestCase
...
     {
-        $this->orderItemMock = $this->createPartialMock(
-            Item::class,
-            ['getChildrenItems', 'isDummy', 'getHasChildren', 'getId', 'getParentItemId']
-        );
+        $this->orderItemMock = $this->getMockBuilder(Item::class)
+            ->disableOriginalConstructor()
+            ->disableOriginalClone()
+            ->disableArgumentCloning()
+            ->disallowMockingUnknownTypes()
```
Comment (lbajsarowicz) [inline app/code/Magento/Sales/Test/Unit/Model/Order/CreditmemoFactoryTest.php]: **YAGNI** If original code is not performing any `clone` then you don't need to provide these methods.

File: lib/internal/Magento/Framework/Pricing/Test/Unit/Adjustment/CollectionTest.php [hunk 5]
```diff
@@ -92,14 +92,15 @@ public function getItemsDataProvider()
             [['adj1'], ['adj1']],
             [['adj4'], ['adj4']],
             [['adj1', 'adj4'], ['adj1', 'adj4']],
-            [['adj1', 'adj2', 'adj3', 'adj4'], ['adj3', 'adj1', 'adj2', 'adj4']]
+            [['adj1', 'adj2', 'adj3', 'adj4'], ['adj3', 'adj1', 'adj2', 'adj4']],
```
Comment (lbajsarowicz) [inline lib/internal/Magento/Framework/Pricing/Test/Unit/Adjustment/CollectionTest.php]: Is that change necessary? :-)

**Review by lbajsarowicz — CHANGES_REQUESTED** [review]
Thank you for your contribution. My only concern is the code that is probably not necessary (You Ain't Gonna Need It). I really appreciate that you want to improve PHPUnit tests!

**Comment by AntonEvers** [comment]
Thanks @lbajsarowicz for reviewing this. All comments are processed.

**Comment by AntonEvers** [comment]
@magento run all tests

**Comment by lbajsarowicz** [comment]
We need to wait for Tests to pass, and then I'm going to approve the PR.

**Review by ihor-sviziev — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-7874](https://jira.corp.magento.com/browse/ENGCOM-7874) has been created to process this Pull Request

**Comment by engcom-Alfa** [comment]
QA not applicable

**Review by lbajsarowicz — APPROVED** [review]
:heavy_check_mark: Thanks!

**Comment by magento-engcom-team** [comment]
Hi @lbajsarowicz, thank you for the review.
[ENGCOM-7874](https://jira.corp.magento.com/browse/ENGCOM-7874) has been created to process this Pull Request

**Comment by engcom-Alfa** [comment]
@magento create issue

**Pull request merged**